app/windows/PublishToIPNS/renderer.jsx

```diff
@@ -0,0 +1,86 @@
+import React from 'react'
+import ReactDom from 'react-dom'
+
+import { remote } from 'electron'
+import { initIPFSClient, publishToIPNS, getPeer } from '../../api'
+import { trackEvent } from '../../stats'
+
+// Load Components
+import {
+  Window,
+  Content,
+  Input,
+  Toolbar,
+  Actionbar,
+  Button
+} from 'react-photonkit'
+import ProgressOverlay from '../../components/ProgressOverlay'
+import CircularProgress from '../../components/CircularProgress'
+
+class PublishToIPNSWindow extends React.Component {
+  state = {
+    loading: false
+  }
+
+  componentDidMount () {
+    trackEvent('PublishToIPNSWindowOpen')
+    getPeer().then(peer => this.setState({ peer }))
+  }
+
+  handleSubmit = (event) => {
+    event.preventDefault()
+    this.setState({ loading: true })
+    const hash = event.target[1].value
+    publishToIPNS(hash)
+      .then(result => {
+        this.setState({ loading: false })
+        const message = `Successfully published ${result.value} to IPNS!`
+        remote.dialog.showMessageBox({ type: 'info', message, cancelId: 0, buttons: ['Ok'] })
+        window.close()
+      })
+      .catch(err => {
+        this.setState({ loading: false })
+        remote.dialog.showErrorBox('Error', err.message)
+      })
+  }
+
+  render () {
+    const { loading, peer } = this.state
+
+    return (
+      <form onSubmit={this.handleSubmit}>
+        <Window>
+          <Content>
+            <Input
+              label="Your peer ID"
+              type="text"
+              value={peer ? peer.id : 'Loading...'}
+              readOnly
```

Comment:
For some reason I cannot copy this. Should we add a button to copy the ID in the clipboard?
(Same issue goes in the settings)